wp-includes/components/class-wp-service-worker-uploaded-image-caching-component.php

```diff
@@ -0,0 +1,118 @@
+<?php
+/**
+ * WP_Service_Worker_Uploaded_Image_Caching_Component class.
+ *
+ * @package PWA
+ */
+
+/**
+ * Use a cache-first caching strategy for uploaded images.
+ *
+ * @since 0.6
+ */
+class WP_Service_Worker_Uploaded_Image_Caching_Component implements WP_Service_Worker_Component {
```

Comment:
I guess it makes sense to start with images here since they're by far the most common. What about e.g. audio though? I assume video would be too resource-heavy.

Reply:
Yeah, that's right. And a reason against doing audio/video is that they add the additional complexity of browsers performing range requests to get back data. There are a few wrinkles for serving cached audio/video: https://developers.google.com/web/tools/workbox/guides/advanced-recipes#cached-av

This being the case, it seems best to leave audio/video caching to be left up to a site to opt-in to.